src/sentry/utils/performance_issues/performance_detection.py

```diff
@@ -525,25 +526,31 @@ def visit_span(self, span: Span):
         if not fingerprint:
             return
 
+        hash = span.get("hash", None)
+        if not hash:
+            return
```

Comment:
Have you considered replacing the broad `fingerprint` on the op w/ just the hash? Then we can do  
```py
most_recent_start =  self.most_recent_start_time[hash]
delta_to_previous_span_start_time =  span["start_timestamp"] - most_recent_start
```
Then each span would look up it's last seen similar span time.

Reply:
@k-fish N+1 API calls only applies to one span op anyway, so I don't need fingerprinting at all. I could just use a single array, and single values! I opted to keep the code similar to the other detectors, for readability, so I did the fingerprinting thing. What do you think, is the hash thing worth it, or is it better to keep the detectors looking similar for now?

Reply:
We definitely should be using the `span_group` hash for all of our detectors moving forward and just improve that logic to help both perf issues and suspect spans, if we have good ideas for better detecting 'similarity'

Reply:
Makes sense!